salt/pillar/http_json.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+"""
+A module that adds data to the Pillar structure retrieved by an http request
+
+
+Configuring the HTTP_JSON ext_pillar
+====================================
+
+Set the following Salt config to setup Foreman as external pillar source:
+
+.. code-block:: json
+
+  ext_pillar:
+    - http_json:
+        url: http://example.com/api/minion_id
+        ::TODO::
+        username: username
+        password: password
+
+Module Documentation
+====================
+"""
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+
+
+def ext_pillar(minion_id,
+               pillar,  # pylint: disable=W0613
+               url=None):
```

Comment:
Since I copied that part from the http_yaml.py  I am not certain, but I would guess that it would so that the structure would be easy to differentiate between the options when the username and password were added in. But I can't think of a case where it makes sense for url to be None.
I really don't care which one it is. The one thing on the side keeping it a kwarg is that it would be nice to keep http_json and http_yaml consistent. I wouldn't want to change that one as well and risk breaking people that have started using it. Granted it probably isn't many people since it was introduced in 2016.3. @t0rrant who wrote http_yaml.py